drivers/wifi/wfx200/wfx200_host.c

```diff
@@ -0,0 +1,730 @@
+	}
+	return SL_STATUS_OK;
+}
+
+sl_status_t sl_wfx_host_free_buffer(void *buffer, sl_wfx_buffer_type_t type)
+{
+	ARG_UNUSED(type);
+
+	if (buffer != NULL) {
+		k_heap_free(&wfx200_0->heap, buffer);
+	}
+	return SL_STATUS_OK;
+}
+
+/**
+ * Bus access
+ */
+sl_status_t sl_wfx_host_lock(void)
+{
+	/* Time taken from silabs sample code */
+	if (k_mutex_lock(&wfx200_0->bus_mutex, K_MSEC(500)) == -EAGAIN) {
+		LOG_DBG("Wi-Fi driver mutex timeout");
+		return SL_STATUS_TIMEOUT;
+	}
+	return SL_STATUS_OK;
+}
+
+sl_status_t sl_wfx_host_unlock(void)
+{
+	k_mutex_unlock(&wfx200_0->bus_mutex);
+	return SL_STATUS_OK;
+}
+
+/* WF200 host bus API */
+sl_status_t sl_wfx_host_init_bus(void)
+{
+	/* Initialization happens in wfx200_init, nothing to do here */
+	return SL_STATUS_OK;
+}
+
+sl_status_t sl_wfx_host_deinit_bus(void)
+{
+	/* Nothing to do here */
+	return SL_STATUS_OK;
+}
+
+sl_status_t sl_wfx_host_enable_platform_interrupt(void)
+{
+	wfx200_enable_interrupt(wfx200_0);
+	return SL_STATUS_OK;
+}
+
+sl_status_t sl_wfx_host_disable_platform_interrupt(void)
+
+{
+	wfx200_disable_interrupt(wfx200_0);
+	return SL_STATUS_OK;
+}
+
+/* WF200 host SPI bus API */
+sl_status_t sl_wfx_host_spi_cs_assert(void)
+{
+	/* Not needed, cs will be handled by spi driver */
+	return SL_STATUS_OK;
+}
+
+sl_status_t sl_wfx_host_spi_cs_deassert(void)
+{
+	/* Not needed, cs will be handled by spi driver */
+	return SL_STATUS_OK;
+}
+
+/* Despite its name cs will be handled by spi hardware */
+sl_status_t sl_wfx_host_spi_transfer_no_cs_assert(sl_wfx_host_bus_transfer_type_t type,
+						  uint8_t *header,
+						  uint16_t header_length,
+						  uint8_t *buffer,
+						  uint16_t buffer_length)
+{
+	const struct wfx200_config *config = wfx200_0->dev->config;
+	int err;
+	const struct spi_buf_set tx = {
+		.buffers = (const struct spi_buf[]){ {
+							     .buf = header,
+							     .len = header_length
+						     }, {
+							     .buf = buffer,
+							     .len = buffer_length
+						     }, },
+		.count = (type == SL_WFX_BUS_WRITE)?2:1,
+	};
+	const struct spi_buf_set rx = {
+		.buffers = (const struct spi_buf[]){ {
+							     .buf = NULL,
+							     .len = header_length
+						     }, {
+							     .buf = buffer,
+							     .len = buffer_length
+						     }, },
+		.count = 2,
+	};
+
+	if (header == NULL || buffer == NULL) {
+		return SL_STATUS_FAIL;
+	}
+	if (type == SL_WFX_BUS_WRITE) {
+		err = spi_write_dt(&config->bus, &tx);
+		if (err < 0) {
+			LOG_ERR("spi_write fail: %d", err);
+			return SL_STATUS_FAIL;
+		}
+	} else {
+		err = spi_transceive_dt(&config->bus, &tx, &rx);
+		if (err < 0) {
+			LOG_ERR("spi_transceive fail: %d", err);
+			return SL_STATUS_FAIL;
+		}
+	}
+	return SL_STATUS_OK;
+}
+
+#define TRUNCATED_MESSAGE "<truncated>"
+BUILD_ASSERT(CONFIG_WIFI_WFX200_LOG_BUFFER_SIZE > sizeof(TRUNCATED_MESSAGE),
+	     "Log buffer must allow \"" TRUNCATED_MESSAGE "\" message");
+
+void sl_wfx_host_log(const char *string, ...)
```

Comment:
So that this function is called, correspondent [#define should be set in Silicon Labs driver](https://github.com/SiliconLabs/wfx-fullMAC-driver/blob/master/wfx_fmac_driver/sl_wfx_constants.h#L168)

Would it make sense to wrap it in a KConfig?
Because changing the Silicon Labs part or providing define via e.g. CMake is not really Zephyr-ish way.

Reply:
I do not understand your complaint here.
CONFIG_WIFI_WFX200_LOG_BUFFER_SIZE is a KConfig variable.

Reply:
@rvoell-gcx  I'm not referring to the log buffer size.
I'm referring to SL_WFX_DEBUG_MASK